zerver/lib/actions.py

```diff
@@ -4921,24 +4922,33 @@ def notify_user_update_custom_profile_data(user_profile: UserProfile,
                                            field: Dict[str, Union[int, str, List[int], None]]) -> None:
     if field['type'] == CustomProfileField.USER:
         field_value = ujson.dumps(field['value'])  # type: Union[int, str, List[int], None]
+        field_rendered_value = None
     else:
         field_value = field['value']
+        field_rendered_value = field['rendered_value']
     payload = dict(user_id=user_profile.id, custom_profile_field=dict(id=field['id'],
-                                                                      value=field_value))
+                                                                      value=field_value,
+                                                                      rendered_value=field_rendered_value))
     event = dict(type="realm_user", op="update", person=payload)
     send_event(user_profile.realm, event, active_user_ids(user_profile.realm.id))
 
 def do_update_user_custom_profile_data(user_profile: UserProfile,
                                        data: List[Dict[str, Union[int, str, List[int]]]]) -> None:
     with transaction.atomic():
         for field in data:
-            field_value, created = CustomProfileFieldValue.objects.update_or_create(
+            field_value, created = CustomProfileFieldValue.objects.get_or_create(
                 user_profile=user_profile,
-                field_id=field['id'],
-                defaults={'value': field['value']})
+                field_id=field['id'])
+            field_value.value = field['value']
+            if field_value.field.is_renderable():
+                field_value.rendered_value = bugdown_convert(str(field['value']))
```

Comment:
`field['value']` will be already `str` in case of Short and Long text, but I assume that `mypy` throws error if you don't convert it to str, as `field['value']` type is generic `Union[int, str, List[int]]`?

Reply:
Precisely. I was uncomfortable with using str(), but I didn't have too much choice here because of mypy.